File: Structural/Bridge/Python/Structure/RefinedAbstraction.py
"""
Bridge Pattern - Refined Abstraction
Extends the abstraction without changing the implementor.
"""

from Abstraction import Abstraction


class RefinedAbstraction(Abstraction):
    """
    You can extend the Abstraction without changing the Implementor classes.
    This allows for variations in the abstraction without affecting the
    implementation hierarchy.
    """
    
    def operation(self) -> str:
        """
        Extended operation that builds upon the base abstraction.
        
        Returns:
            str: Result of the refined operation
        """
        return f"RefinedAbstraction: Extended operation with:\n{self._implementor.operation_impl()}"
    
    def another_operation(self) -> str:
        """
        Additional operation specific to the refined abstraction.
        
        Returns:
            str: Result of another operation
        """
        return f"RefinedAbstraction: Another operation with:\n{self._implementor.operation_impl()}"

File: Structural/Bridge/Java/Structure/RefinedAbstraction.java
// RefinedAbstraction adds more behavior but still delegates to the Implementor
public class RefinedAbstraction extends Abstraction {
    public RefinedAbstraction(Implementor implementor) {
        super(implementor);
    }

    public void enhancedOperation(String prefix, String payload) {
        System.out.println("RefinedAbstraction: enhanced operation (" + prefix + ") -> " + payload);
        implementor.operationImpl(prefix + ":" + payload);
    }
}
